contracts/zap/staking/SolidZapStaker.sol

```diff
@@ -278,6 +282,10 @@ contract SolidZapStaker is BaseSolidZapStaker {
             token1Address
         );
 
+        if (token0BalanceAfter <= token0BalanceBefore || token1BalanceAfter <= token1BalanceBefore) {
```

Comment:
In case of single swap `inputToken` is one of `token0Address`, `token1Address`. So `inputToken` balance after will be lesser compared to before, because it will part of it will be swapped. Do I miss something?

Reply:
the balances refer to the zap contract's balance of token0/token1

Steps:
0) the 'before' balances are 0 initially (or any number if someone foolishly locked any token0/token1 in this contract)
1) we transfer the input token from user to the zap contract (e.g. transfer 100 token0 to the zap contract)
2) we do the swap (e.g. use 50 token0 to obtain 50 token1)
3) the 'after' balances are 50 token0 and 50 token1

Reply:
Thanks for explanation, I incorrectly read the code. Everything is fine here 🙂